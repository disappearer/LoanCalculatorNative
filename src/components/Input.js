import React from 'react';
import * as _ from 'lodash';
import PropTypes from 'prop-types';
import { Slider, View, Picker, Text } from 'react-native';

const Input = ({ label, min, max, step, value, onChange }) => (
  <View>
    <Text style={{ color: 'green' }}>{label}</Text>
    <Slider
      type="range"
      minimumValue={min}
      maximumValue={max}
      step={step}
      value={value}
      onValueChange={onChange}
    />
    <Picker selectedValue={value} onValueChange={onChange}>
      {_.range(min, max + 1, step).map(itemValue => (
        <Picker.Item
          key={itemValue}
          value={itemValue}
          label={itemValue.toString()}
        />
      ))}
    </Picker>
  </View>
);

Input.propTypes = {
  label: PropTypes.string.isRequired,
  min: PropTypes.number.isRequired,
  max: PropTypes.number.isRequired,
  step: PropTypes.number.isRequired,
  onChange: PropTypes.func.isRequired
};

export default Input;
