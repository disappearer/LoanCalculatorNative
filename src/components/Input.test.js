import React from 'react';
import enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import Input from './Input';

enzyme.configure({ adapter: new Adapter() });

describe('Input component', () => {
  it('When not fetching renders as expected', () => {
    const wrapper = shallow(
      <Input
        isFetching={false}
        label="Total Amount"
        min={1}
        max={6}
        step={1}
        value={3}
        onChange={() => {}}
      />
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('When fetching renders as expected', () => {
    const wrapper = shallow(
      <Input
        isFetching={true}
        label="Total Amount"
        min={1}
        max={6}
        step={1}
        value={3}
        onChange={() => {}}
      />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
