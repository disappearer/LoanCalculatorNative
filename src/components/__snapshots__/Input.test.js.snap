// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input component When fetching renders as expected 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Input
    isFetching={true}
    label="Total Amount"
    max={6}
    min={1}
    onChange={[Function]}
    step={1}
    value={3}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "green",
            }
          }
        >
          Total Amount
        </Text>,
        <Slider
          disabled={false}
          maximumValue={6}
          minimumValue={1}
          onValueChange={[Function]}
          step={1}
          type="range"
          value={3}
        />,
        <Picker
          mode="dialog"
          onValueChange={[Function]}
          selectedValue={3}
        >
          <PickerItem
            label="1"
            value={1}
          />
          <PickerItem
            label="2"
            value={2}
          />
          <PickerItem
            label="3"
            value={3}
          />
          <PickerItem
            label="4"
            value={4}
          />
          <PickerItem
            label="5"
            value={5}
          />
          <PickerItem
            label="6"
            value={6}
          />
        </Picker>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "accessible": true,
          "allowFontScaling": true,
          "children": "Total Amount",
          "ellipsizeMode": "tail",
          "style": Object {
            "color": "green",
          },
        },
        "ref": null,
        "rendered": "Total Amount",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "disabled": false,
          "maximumValue": 6,
          "minimumValue": 1,
          "onValueChange": [Function],
          "step": 1,
          "type": "range",
          "value": 3,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <PickerItem
              label="1"
              value={1}
            />,
            <PickerItem
              label="2"
              value={2}
            />,
            <PickerItem
              label="3"
              value={3}
            />,
            <PickerItem
              label="4"
              value={4}
            />,
            <PickerItem
              label="5"
              value={5}
            />,
            <PickerItem
              label="6"
              value={6}
            />,
          ],
          "mode": "dialog",
          "onValueChange": [Function],
          "selectedValue": 3,
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "1",
            "nodeType": "class",
            "props": Object {
              "label": "1",
              "value": 1,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "2",
            "nodeType": "class",
            "props": Object {
              "label": "2",
              "value": 2,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "3",
            "nodeType": "class",
            "props": Object {
              "label": "3",
              "value": 3,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "4",
            "nodeType": "class",
            "props": Object {
              "label": "4",
              "value": 4,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "5",
            "nodeType": "class",
            "props": Object {
              "label": "5",
              "value": 5,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "6",
            "nodeType": "class",
            "props": Object {
              "label": "6",
              "value": 6,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "green",
              }
            }
          >
            Total Amount
          </Text>,
          <Slider
            disabled={false}
            maximumValue={6}
            minimumValue={1}
            onValueChange={[Function]}
            step={1}
            type="range"
            value={3}
          />,
          <Picker
            mode="dialog"
            onValueChange={[Function]}
            selectedValue={3}
          >
            <PickerItem
              label="1"
              value={1}
            />
            <PickerItem
              label="2"
              value={2}
            />
            <PickerItem
              label="3"
              value={3}
            />
            <PickerItem
              label="4"
              value={4}
            />
            <PickerItem
              label="5"
              value={5}
            />
            <PickerItem
              label="6"
              value={6}
            />
          </Picker>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "accessible": true,
            "allowFontScaling": true,
            "children": "Total Amount",
            "ellipsizeMode": "tail",
            "style": Object {
              "color": "green",
            },
          },
          "ref": null,
          "rendered": "Total Amount",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "disabled": false,
            "maximumValue": 6,
            "minimumValue": 1,
            "onValueChange": [Function],
            "step": 1,
            "type": "range",
            "value": 3,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <PickerItem
                label="1"
                value={1}
              />,
              <PickerItem
                label="2"
                value={2}
              />,
              <PickerItem
                label="3"
                value={3}
              />,
              <PickerItem
                label="4"
                value={4}
              />,
              <PickerItem
                label="5"
                value={5}
              />,
              <PickerItem
                label="6"
                value={6}
              />,
            ],
            "mode": "dialog",
            "onValueChange": [Function],
            "selectedValue": 3,
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "1",
              "nodeType": "class",
              "props": Object {
                "label": "1",
                "value": 1,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": "2",
              "nodeType": "class",
              "props": Object {
                "label": "2",
                "value": 2,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": "3",
              "nodeType": "class",
              "props": Object {
                "label": "3",
                "value": 3,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": "4",
              "nodeType": "class",
              "props": Object {
                "label": "4",
                "value": 4,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": "5",
              "nodeType": "class",
              "props": Object {
                "label": "5",
                "value": 5,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": "6",
              "nodeType": "class",
              "props": Object {
                "label": "6",
                "value": 6,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`Input component When not fetching renders as expected 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Input
    isFetching={false}
    label="Total Amount"
    max={6}
    min={1}
    onChange={[Function]}
    step={1}
    value={3}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "green",
            }
          }
        >
          Total Amount
        </Text>,
        <Slider
          disabled={false}
          maximumValue={6}
          minimumValue={1}
          onValueChange={[Function]}
          step={1}
          type="range"
          value={3}
        />,
        <Picker
          mode="dialog"
          onValueChange={[Function]}
          selectedValue={3}
        >
          <PickerItem
            label="1"
            value={1}
          />
          <PickerItem
            label="2"
            value={2}
          />
          <PickerItem
            label="3"
            value={3}
          />
          <PickerItem
            label="4"
            value={4}
          />
          <PickerItem
            label="5"
            value={5}
          />
          <PickerItem
            label="6"
            value={6}
          />
        </Picker>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "accessible": true,
          "allowFontScaling": true,
          "children": "Total Amount",
          "ellipsizeMode": "tail",
          "style": Object {
            "color": "green",
          },
        },
        "ref": null,
        "rendered": "Total Amount",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "disabled": false,
          "maximumValue": 6,
          "minimumValue": 1,
          "onValueChange": [Function],
          "step": 1,
          "type": "range",
          "value": 3,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <PickerItem
              label="1"
              value={1}
            />,
            <PickerItem
              label="2"
              value={2}
            />,
            <PickerItem
              label="3"
              value={3}
            />,
            <PickerItem
              label="4"
              value={4}
            />,
            <PickerItem
              label="5"
              value={5}
            />,
            <PickerItem
              label="6"
              value={6}
            />,
          ],
          "mode": "dialog",
          "onValueChange": [Function],
          "selectedValue": 3,
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "1",
            "nodeType": "class",
            "props": Object {
              "label": "1",
              "value": 1,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "2",
            "nodeType": "class",
            "props": Object {
              "label": "2",
              "value": 2,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "3",
            "nodeType": "class",
            "props": Object {
              "label": "3",
              "value": 3,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "4",
            "nodeType": "class",
            "props": Object {
              "label": "4",
              "value": 4,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "5",
            "nodeType": "class",
            "props": Object {
              "label": "5",
              "value": 5,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "6",
            "nodeType": "class",
            "props": Object {
              "label": "6",
              "value": 6,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "green",
              }
            }
          >
            Total Amount
          </Text>,
          <Slider
            disabled={false}
            maximumValue={6}
            minimumValue={1}
            onValueChange={[Function]}
            step={1}
            type="range"
            value={3}
          />,
          <Picker
            mode="dialog"
            onValueChange={[Function]}
            selectedValue={3}
          >
            <PickerItem
              label="1"
              value={1}
            />
            <PickerItem
              label="2"
              value={2}
            />
            <PickerItem
              label="3"
              value={3}
            />
            <PickerItem
              label="4"
              value={4}
            />
            <PickerItem
              label="5"
              value={5}
            />
            <PickerItem
              label="6"
              value={6}
            />
          </Picker>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "accessible": true,
            "allowFontScaling": true,
            "children": "Total Amount",
            "ellipsizeMode": "tail",
            "style": Object {
              "color": "green",
            },
          },
          "ref": null,
          "rendered": "Total Amount",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "disabled": false,
            "maximumValue": 6,
            "minimumValue": 1,
            "onValueChange": [Function],
            "step": 1,
            "type": "range",
            "value": 3,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <PickerItem
                label="1"
                value={1}
              />,
              <PickerItem
                label="2"
                value={2}
              />,
              <PickerItem
                label="3"
                value={3}
              />,
              <PickerItem
                label="4"
                value={4}
              />,
              <PickerItem
                label="5"
                value={5}
              />,
              <PickerItem
                label="6"
                value={6}
              />,
            ],
            "mode": "dialog",
            "onValueChange": [Function],
            "selectedValue": 3,
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "1",
              "nodeType": "class",
              "props": Object {
                "label": "1",
                "value": 1,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": "2",
              "nodeType": "class",
              "props": Object {
                "label": "2",
                "value": 2,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": "3",
              "nodeType": "class",
              "props": Object {
                "label": "3",
                "value": 3,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": "4",
              "nodeType": "class",
              "props": Object {
                "label": "4",
                "value": 4,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": "5",
              "nodeType": "class",
              "props": Object {
                "label": "5",
                "value": 5,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": "6",
              "nodeType": "class",
              "props": Object {
                "label": "6",
                "value": 6,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
